sandbox_to_csv.py

```diff
@@ -0,0 +1,258 @@
+                        name. Default: {pilot}_{message_type}_{sdate}_{edate}
+  --json                Supply flag if file is to be exported as newline json
+                        instead of CSV file. Default: False
+  --aws_profile AWS_PROFILE
+                        Supply name of AWS profile if not using default
+                        profile. AWS profile must be configured in
+                        ~/.aws/credentials on your machine. See https://boto3.
+                        amazonaws.com/v1/documentation/api/latest/guide/config
+                        uration.html#shared-credentials-file for more
+                        information.
+  --zip                 Supply flag if output files should be zipped together.
+                        Default: False
+
+"""
+from __future__ import print_function
+from argparse import ArgumentParser
+import boto3
+from botocore.exceptions import ProfileNotFound
+from copy import copy
+import dateutil.parser
+from datetime import datetime, timedelta
+from functools import reduce
+import json
+import os
+import csv
+import threading
+import time
+import traceback
+import zipfile
+
+
+from flattener import load_flattener
+from s3_file_mover import CvPilotFileMover
+
+
+class SandboxExporter(object):
+
+    def __init__(self, bucket='usdot-its-cvpilot-public-data', pilot='wydot',
+                message_type='bsm', sdate=None, edate=None, csv=True, zip=False,
+                output_convention='{pilot}_{message_type}_{sdate}_{edate}',
+                aws_profile="default"):
+        # set up
+        self.bucket = bucket
+        self.pilot = pilot
+        self.message_type = message_type
+        self.sdate = None
+        self.edate = None
+        self.csv = csv
+        self.zip = zip
+        self.output_convention = output_convention
+        self.aws_profile = aws_profile
+
+        if sdate:
+            self.sdate = dateutil.parser.parse(sdate)
+        if edate:
+            self.edate = dateutil.parser.parse(edate)
+        else:
+            self.edate = self.sdate + timedelta(hours=24)
+
+        aws_session = self.create_aws_session()
+        s3botoclient = aws_session.client('s3')
+        self.mover = CvPilotFileMover(target_bucket=bucket,
+                                 source_bucket_prefix="",
+                                 source_key_prefix="",
+                                 validation_queue_name=None,
+                                 log=False,
+                                 s3_client=s3botoclient)
+
+        flattenerMod = load_flattener('{}/{}'.format(pilot, message_type.upper()))
+        self.flattener = flattenerMod()
+        self.current_recs = []
+        self.file_names = []
+
+    def create_aws_session(self):
+        try:
+            session = boto3.session.Session(profile_name=self.aws_profile)
+        except ProfileNotFound:
+            print('Please supply a valid AWS profile name.')
```

Comment:
Since you're using a logger and print_func elsewhere, I would suggest replacing these print statements with that.

Reply:
Modified to use print_func.